documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,208 @@
+Single-Pass Scan
+################
+
+-----------------------------------------------------------
+inclusive_scan Function Template
+-----------------------------------------------------------
+
+The ``inclusive_scan`` function computes the inclusive prefix sum using a given binary operation.
+The function implements a single-pass algorithm, where each input element is read exactly once from
+global memory and each output element is written to exactly once in global memory.
+
+As this algorithm uses inter-work-group communication, it must be run on a device that supports
+strong forward progress guarantees, such as the `Intel® Data Center GPU Max Series
+<https://www.intel.com/content/www/us/en/products/details/discrete-gpus/data-center-gpu/max-series/products.html>`_.
```

Comment:
Forward progress guarantees is a complex topic for GPUs. We should be careful while doing statements about it.

As far as I know (not totally sure), Data Center GPU Max guarantees at least parallel forward progress between work-groups for  [cooperative launches](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_root_group.asciidoc). Normal launches provide weakly parallel forward guarantee (essentially, no guarantees).
At this point, I suppose, one should thoroughly investigate GPU architecture specifics to understand whether this synchronization pattern would work or not. That's why I recommend simply specifying the list of the supported GPUs.

Reply:
This makes sense.  I was hoping there was some better existing reference to point to, but perhaps its best to just list "known" supported GPUs for now.

Reply:
Unfortunately, it's like folklore. At least, I am not aware of such references.

Reply:
Would it be sufficient to remove the phrase "must be run on a device that supports strong forward progress guarantees" and simply state that the algorithm has been designed specifically to run on Intel® Data Center GPU Max Series?

Reply:
I've been thinking about the previous Dan's suggestion and the flexibility of it is a good thing.
Adam, what do you think about that slight correction to your note, which avoids making that general statement about the GPU?

I also mentioned device USM as it is an optional SYCL feature. 

>The algorithm is designed to be compatible with a variety of devices that provide at least parallel forward guarantees between work-groups, due to cross-work-group communication. Additionally, it requires support for device USM. It has been verified to be compatible with Intel® Data Center GPU Max Series.

Reply:
I like your wording a lot. I updated the paragraph accordingly.